.changeset/hungry-games-compete.md

```diff
@@ -0,0 +1,9 @@
+---
+'@tus/file-store': minor
+'@tus/gcs-store': minor
+'@tus/s3-store': minor
+'@tus/utils': minor
+---
+
+Add basic storage information to the Upload model. You can now access `upload.storage`
+which has `type` (`file`, `s3`, `gcs`), `path`, and when applicable `bucket`.
```

Comment:
In tusd I have decided to use `key` for s3/gcs as that is the terminology that the providers also use. Maybe you want to consider this as well.

Reply:
I saw that, but I don't want conditional types and conceptually we can treat them the same, you use it to know where an upload is stored.

Reply:
Ok, sounds good.